lib/main.js

```diff
@@ -41,8 +43,20 @@ function _interpolate (envValue, environment, config) {
 }
 
 function expand (config) {
-  // if ignoring process.env, use a blank object
-  const environment = config.ignoreProcessEnv ? {} : process.env
+  let environment
+
+  switch (config.ignoreProcessEnv) {
+    case 'read-only':
+      environment = rfdc()(process.env)
```

Comment:
Could add a TODO to drop this library and use `structuredClone` when only Node 18 is supported

Reply:
Actually, `process.env` won't be a nested object will it? So this could just use `Object.assign` rather than an external Library?

Reply:
Wasn't aware of `structuredClone` until now - whether we need a TODO for this or not is up to @motdotla.

As for the depth of `process.env`, _theoretically_ it will be flat, but _technically_ one could mutate it beforehand to be more than one level deep, and it's probably a point for tricky bugs if this library wasn't defensive against that.

Reply:
[I stand corrected](https://nodejs.org/api/process.html#process_process_env:~:text=Assigning%20a%20property%20on%20process.env%20will%20implicitly%20convert%20the%20value%20to%20a%20string.%20This%20behavior%20is%20deprecated.%20Future%20versions%20of%20Node.js%20may%20throw%20an%20error%20when%20the%20value%20is%20not%20a%20string%2C%20number%2C%20or%20boolean.) You're absolutely right about not needing the deep clone.